types/errors.go

```diff
@@ -31,8 +33,9 @@ const (
 	CodeUnknownAddress    CodeType = 9
 	CodeInsufficientCoins CodeType = 10
 	CodeInvalidCoins      CodeType = 11
+	CodeGeneric           CodeType = 12
 
-	CodeGenesisParse CodeType = 0xdead // TODO: remove ? // why remove?
+	CodeGenesisParse CodeType = 0xdead
```

Comment:
Included this comment because I wasn't sure about the relation between Tx codes and non-tx codes. Maybe we keep 0-1000 for tx codes, and 1000-2000 for codes for other ABCI messages ? Not sure ...

Reply:
I suggest ErrInternal for all non-tx types.